internal/engine/image_build_and_deployer_test.go

```diff
@@ -536,6 +537,61 @@ func TestDeployInjectsOverrideCommand(t *testing.T) {
 	assert.Empty(t, c.Args)
 }
 
+func (f *ibdFixture) firstPodTemplateSpecHash() k8s.PodTemplateSpecHash {
+	entities, err := k8s.ParseYAMLFromString(f.k8s.Yaml)
+	if err != nil {
+		f.T().Fatal(err)
+	}
+
+	require.Equal(f.T(), 1, len(entities), "expected only one entity. Yaml contained: %s", f.k8s.Yaml)
+
+	d := entities[0].Obj.(*v1.Deployment)
+	return k8s.PodTemplateSpecHash(d.Spec.Template.Labels[k8s.TiltPodTemplateHashLabel])
+}
+
+func TestDeployInjectsPodTemplateSpecHash(t *testing.T) {
+	f := newIBDFixture(t, k8s.EnvGKE)
+	defer f.TearDown()
+
+	manifest := NewSanchoDockerBuildManifest(f)
+
+	resultSet, err := f.ibd.BuildAndDeploy(f.ctx, f.st, buildTargets(manifest), store.BuildStateSet{})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	hash := f.firstPodTemplateSpecHash()
+
+	require.NotEqual(t, hash, "")
```

Comment:
maybe worth it to have `firstPodTemplateSpecHash` assert that the hash is non-empty? 🤷‍♀